core/processor/abci.go

```diff
@@ -1912,6 +1914,20 @@ func (app *App) enactCancelTransfer(ctx context.Context, prop *types.Proposal) {
 	}
 }
 
+func (app *App) enactMarketStateUpdate(ctx context.Context, prop *types.Proposal) {
+	prop.State = types.ProposalStateEnacted
+	changes := prop.Terms.GetMarketStateUpdate().Changes
+	if err := app.exec.VerifyUpdateMarketState(changes); err != nil {
+		app.log.Error("failed to enact governance market state update", logging.String("proposal", prop.ID), logging.String("error", err.Error()))
+		prop.FailWithErr(types.ProporsalErrorFailedGovernanceTransferCancel, err)
```

Comment:
`ProporsalErrorFailedGovernanceTransferCancel` copy and paste error?

Reply:
yes